framework/include/problems/FEProblemBase.h

```diff
@@ -1741,51 +1741,51 @@ class FEProblemBase : public SubProblem, public Restartable
-  PerfID _compute_residual_l2_norm_timer;
-  PerfID _compute_residual_sys_timer;
-  PerfID _compute_residual_internal_timer;
-  PerfID _compute_residual_type_timer;
-  PerfID _compute_transient_implicit_residual_timer;
-  PerfID _compute_residual_tags_timer;
-  PerfID _compute_jacobian_internal_timer;
-  PerfID _compute_jacobian_tags_timer;
-  PerfID _compute_jacobian_blocks_timer;
-  PerfID _compute_bounds_timer;
-  PerfID _compute_post_check_timer;
-  PerfID _compute_damping_timer;
-  PerfID _possibly_rebuild_geom_search_patches_timer;
-  PerfID _initial_adapt_mesh_timer;
-  PerfID _adapt_mesh_timer;
-  PerfID _update_mesh_xfem_timer;
-  PerfID _mesh_changed_timer;
-  PerfID _mesh_changed_helper_timer;
-  PerfID _check_problem_integrity_timer;
-  PerfID _serialize_solution_timer;
-  PerfID _check_nonlinear_convergence_timer;
-  PerfID _check_linear_convergence_timer;
-  PerfID _update_geometric_search_timer;
-  PerfID _exec_multi_apps_timer;
-  PerfID _backup_multi_apps_timer;
+  const PerfID _initial_setup_timer;
+  const PerfID _project_solution_timer;
+  const PerfID _compute_indicators_timer;
+  const PerfID _compute_markers_timer;
+  const PerfID _compute_user_objects_timer;
+  const PerfID _execute_controls_timer;
+  const PerfID _execute_samplers_timer;
+  const PerfID _update_active_objects_timer;
+  const PerfID _reinit_because_of_ghosting_or_new_geom_objects_timer;
+  const PerfID _exec_multi_app_transfers_timer;
+  const PerfID _init_timer;
+  const PerfID _eq_init_timer;
+  const PerfID _solve_timer;
+  const PerfID _check_exception_and_stop_solve_timer;
+  const PerfID _advance_state_timer;
+  const PerfID _restore_solutions_timer;
+  const PerfID _save_old_solutions_timer;
+  const PerfID _restore_old_solutions_timer;
+  const PerfID _output_step_timer;
+  const PerfID _on_timestep_begin_timer;
+  const PerfID _compute_residual_l2_norm_timer;
+  const PerfID _compute_residual_sys_timer;
+  const PerfID _compute_residual_internal_timer;
+  const PerfID _compute_residual_type_timer;
+  const PerfID _compute_transient_implicit_residual_timer;
+  const PerfID _compute_residual_tags_timer;
+  const PerfID _compute_jacobian_internal_timer;
+  const PerfID _compute_jacobian_tags_timer;
+  const PerfID _compute_jacobian_blocks_timer;
+  const PerfID _compute_bounds_timer;
+  const PerfID _compute_post_check_timer;
+  const PerfID _compute_damping_timer;
+  const PerfID _possibly_rebuild_geom_search_patches_timer;
+  const PerfID _initial_adapt_mesh_timer;
+  const PerfID _adapt_mesh_timer;
+  const PerfID _update_mesh_xfem_timer;
+  const PerfID _mesh_changed_timer;
+  const PerfID _mesh_changed_helper_timer;
+  const PerfID _check_problem_integrity_timer;
+  const PerfID _serialize_solution_timer;
+  const PerfID _check_nonlinear_convergence_timer;
+  const PerfID _check_linear_convergence_timer;
+  const PerfID _update_geometric_search_timer;
+  const PerfID _exec_multi_apps_timer;
+  const PerfID _backup_multi_apps_timer;
```

Comment:
Is it better to have a class to encapsulate all these variables?

Reply:
I made a similar argument when this originally went in, but this is the only place where there are so many. There's really no additional benefit other than moving these declarations out of this class. I just went through and marked them all const while I was doing the "const" treatment for FEProblemBase. It's a separate commit but unrelated to this PR. Just ignore it.